src/main/groovy/com/google/gcloud/task/GCloudTask.groovy

```diff
@@ -1,45 +1,118 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 package com.google.gcloud.task
 
-import com.google.gcloud.GCloudPluginExtension
-import com.google.gcloud.wrapper.GCloud
-import com.google.gcloud.wrapper.GCloudCommandBuilder
-import com.google.gcloud.wrapper.impl.GCloudCommandLineWrapper
+import com.google.gcloud.exec.GradleExecProcessRunner
+import com.google.gcloud.exec.ProcessRunner
+import com.google.gcloud.exec.command.CommandBuilder
 import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
 import org.gradle.api.Incubating
-import org.gradle.api.tasks.Input
+import org.gradle.api.logging.LogLevel
 import org.gradle.api.tasks.TaskAction
 
 /**
- * GCloud task is never exposed, but users can override it if they so please
+ * GCloud raw task is never exposed, but users can override it if they so please
  */
 @Incubating
 class GCloudTask extends DefaultTask {
-    boolean isRunAsync = false
-    @Input String[] command
+
+    static final String GCLOUD_HOME_ENV_VAR_KEY = 'GCLOUD_HOME'
+    static final String GCLOUD_HOME_SYS_PROP_KEY = 'gcloud.home'
+    final CommandBuilder commandBuilder = new CommandBuilder()
+
+    // inputs
+    List<String> command
+    Map<String, ?> args
 
     @TaskAction
-    protected void executeTask() {
-        GCloud gcloud = new GCloudCommandLineWrapper(getCommand())
-        GCloudPluginExtension gcEx = project.extensions.getByName("gcloud")
-        if (gcEx.gcloudHome) {
-            gcloud.setGCloudHome(new File(gcEx.gcloudHome));
+    protected void start() {
+        validateCloudSdk()
+        executeTask()
+    }
+
+    void validateCloudSdk() {
+        // TODO: validate the cloud sdk install somehow here
+    }
+
+    // automatically find the gcloud executable
+    private String getGCloudCommand() {
+        def gcloudHomeSysProp = System.getProperty(GCLOUD_HOME_SYS_PROP_KEY)
+        if (gcloudHomeSysProp) {
+            File gcloudBin = new File(gcloudHomeSysProp, "bin/gcloud")
```

Comment:
Valentin said that windows adds the appropriate extension and this should probably work.  I'll have to do some windows testing on this to see what really happens with gradle wrapping things.